### PR TITLE
Using Stream Builder to deserialization of int[]

### PR DESCRIPTION
**Pull request by Shounaks**
Issue: #7 
we can use IntStream.Builder and iteratively add tokens. I don't know if its wasteful or not... 😥

### CODE REVIEW

File: jr-objects/src/test/java/com/fasterxml/jackson/jr/failing/ReadIntArray7Test.java [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-
 import com.fasterxml.jackson.jr.ob.JSON;
 import com.fasterxml.jackson.jr.ob.TestBase;
 
-public class ReadIntArray7Test extends TestBase
-{
-    public void testReadIntArray() throws Exception
-    {
-        final int[] input = new int[] { 1, 2, 3 };
+import static org.junit.Assert.assertArrayEquals;
```
Comment (Shounaks) [inline jr-objects/src/test/java/com/fasterxml/jackson/jr/failing/ReadIntArray7Test.java:6]: Move to passing.

File: jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/SimpleValueReader.java [hunk 2]
```diff
@@ -1,20 +1,23 @@
 package com.fasterxml.jackson.jr.ob.impl;
 
-import static com.fasterxml.jackson.jr.ob.impl.ValueWriterLocator.*;
+import com.fasterxml.jackson.core.JsonParser;
```
Comment (cowtowncoder) [inline jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/SimpleValueReader.java:3]: As per https://github.com/FasterXML/jackson/blob/master/contribution/jackson-coding-style-guide.md

(written fairly recently, to explain logic of things like import order)

import order needs to start with most general (JDK) and then to more specific (3rd party deps, core Jackson deps, finally project itself).

So let's try not to reorder these (I know IDEs like reordering things when making changes).

File: jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/SimpleValueReader.java [hunk 3]
```diff
@@ -290,10 +293,17 @@ protected Path _readPath(JsonParser p) throws IOException {
         }
     }
 
-    protected int[] _readIntArray(JsonParser p) throws IOException
-    {
-        // !!! TODO
-        throw new JSONObjectException("Reading of int[] not yet implemented");
+    protected int[] _readIntArray(JsonParser p) throws IOException {
+        if (p.getCurrentToken().equals(JsonToken.START_ARRAY)) {
```
Comment (cowtowncoder) [inline jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/SimpleValueReader.java:297]: `getCurrentToken()` is deprecated, let's use `currentToken()`.

File: jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/SimpleValueReader.java [hunk 4]
```diff
@@ -290,10 +293,17 @@ protected Path _readPath(JsonParser p) throws IOException {
...
+            p.nextToken();
+        }
+
+        final IntStream.Builder builder = IntStream.builder();
+        while (!p.getCurrentToken().equals(JsonToken.END_ARRAY)) {
+            builder.add(p.getValueAsInt());
```
Comment (cowtowncoder) [inline jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/SimpleValueReader.java:303]: I wonder what this does for `START_OBJECT` or `START_ARRAY`? We should probably do some other validation.
I'll have to think about this.

File: jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/SimpleValueReader.java (hunk 2 above)
Comment (cowtowncoder) [reply jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/SimpleValueReader.java:3]: np. Sorry, I use Eclipse for OSS work. But if someone else has formatter I'd be happy to add to (ask others to add) formatter on that page.

@JooHyukKim do you happen to have a formatter to share?

**Comment by cowtowncoder** [comment]
Ok, changed decoding logic to avoid StackOverflowException, made it bit stricter.
I think the failing test is wrong in that it has "one too many levels"; where `int` is expected we should not accept Arrays, Objects etc.

**Review by cowtowncoder — APPROVED** [review]

**Pull request merged**